### PR TITLE
[protocol] Include goal output type in goals request

### PR DESCRIPTION
**Pull request by ejgallego**
This is a step to make life of clients easier, cc: @bhaktishh

### CODE REVIEW

File: controller/lsp_core.ml [hunk 1]
```diff
@@ -265,7 +267,26 @@ let do_position_request ~postpone ~params ~handler =
     (Request.Data.PosRequest { uri; handler; point; version; postpone })
 
 let do_hover = do_position_request ~postpone:false ~handler:Rq_hover.hover
-let do_goals = do_position_request ~postpone:true ~handler:Rq_goals.goals
+
+(* We get the format from the params *)
+let get_pp_format_from_config () =
+  match !Fleche.Config.v.pp_type with
+  | 1 -> Rq_goals.Pp
+  | _ -> Rq_goals.Str
```
Comment (Alizter) [inline controller/lsp_core.ml:275]: Do we want to do this or fail on pp types we don't know. I thinking people might use this to see if pp is supported, but it would give a false positive which might cause headaches.

Comment (Alizter) [reply controller/lsp_core.ml:275]: At the very least we should output a warning in the log.

Comment (ejgallego) [reply controller/lsp_core.ml:275]: Yes, it is a pity using deriving_yojson on such unitary types doesn't work as we'd like, hopefully we can improve this part of the serialization.

File: editor/code/src/goals.ts [hunk 2]
```diff
@@ -1,4 +1,12 @@
-import { Position, Uri, WebviewPanel, window, ViewColumn, extensions, commands } from "vscode";
+import {
```
Comment (Alizter) [inline editor/code/src/goals.ts:1]: This appears to be unrelated

Comment (ejgallego) [reply editor/code/src/goals.ts:1]: Yes it was a formatting problem from previous PR, my bad.

It is too tiny to split IMO, so if you don't mind I'll push it here.
